pepper-apis/src/main/java/org/broadinstitute/ddp/route/UserRegistrationRoute.java

```diff
@@ -102,18 +104,26 @@ public Object handle(Request request, Response response, UserRegistrationPayload
         return TransactionWrapper.withTxn(handle -> {
             auth0UserId.set(payload.getAuth0UserId());
             String auth0Domain = payload.getAuth0Domain();
+
+            StudyDto study = null;
             if (doLocalRegistration && StringUtils.isBlank(auth0Domain)) {
-                auth0Domain = ConfigManager.getInstance().getConfig().getConfig(ConfigFile.AUTH0).getString(ConfigFile.DOMAIN);
-                LOG.info("Using auth0 domain {} for local registration", auth0Domain);
+                study = handle.attach(JdbiUmbrellaStudy.class).findByStudyGuid(studyGuid);
+            } else {
+                study = handle.attach(JdbiUmbrellaStudy.class).findByDomainAndStudyGuid(auth0Domain, studyGuid);
             }
-
-            StudyDto study = handle.attach(JdbiUmbrellaStudy.class).findByDomainAndStudyGuid(auth0Domain, studyGuid);
             if (study == null) {
                 ApiError err = new ApiError(ErrorCodes.STUDY_NOT_FOUND, "Could not find study with guid " + studyGuid);
                 LOG.warn(err.getMessage());
                 throw ResponseUtil.haltError(response, HttpStatus.SC_NOT_FOUND, err);
             }
 
+            if (doLocalRegistration && StringUtils.isBlank(auth0Domain)) {
+                auth0Domain = handle.attach(JdbiAuth0Tenant.class).findById(study.getAuth0TenantId())
```

Comment:
As per current database design/constraints, I don't believe queried auth0Domain can be null but to be future proof/extra cautious should we check for null and throw exception ?

Reply:
I think auth0Domain is pretty crucial to our current model, so I don't think it will be made null. Also, this is only in local registration codepath so I'm not too worried about that.